src/odata-to-abstract-sql.ts

```diff
@@ -298,10 +300,17 @@ export const isBindReference = (maybeBind: {
 	);
 };
 
-const isDynamicResource = (resource: Resource): boolean => {
+const isDynamicResource = (
+	resource: Resource,
+	alreadyComputedFieldsLookup: AlreadyComputedFieldsLookup,
+): boolean => {
 	return (
 		resource.definition != null ||
-		resource.fields.some((f) => f.computed != null)
+		resource.fields.some(
+			(f) =>
+				f.computed != null &&
+				!alreadyComputedFieldsLookup[resource.name + f.fieldName],
```

Comment:
It's good to add some form of separator to further reduce the chances of accidental clashes, eg
```suggestion
				!alreadyComputedFieldsLookup[resource.name + '$' + f.fieldName],
```